finetune/scheduler.py

```diff
@@ -1,17 +1,54 @@
 import gc
 import logging
+import functools
+
+import psutil
 
 import tensorflow as tf
 from finetune.base import BaseModel
+from finetune.errors import FinetuneSchedulerError
 
 LOGGER = logging.getLogger("finetune")
 
+
 def bytes_to_meg(x):
     return x / 1024 / 1024
 
+
+def scheduled(fn):
+    @functools.wraps(fn)
+    def scheduled_predict(self, model_file, x, *args, **kwargs):
+        model = self._rotate_in_model(model_file)
+        try:
+            preds = fn(self, model_file=model_file, x=x, *args, model=model, **kwargs)
```

Comment:
Trying to think if there are ever any "user error" style scenarios w/ bad inputs that might cause someone to purge everyone elses models.